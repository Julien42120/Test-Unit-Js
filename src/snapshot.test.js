import React from 'react'
import '@testing-library/jest-dom/extend-expect'


const FlightService = {
  list(){
    return {
      "arrivalDatetime": "2018-06-15T21:00:00",
      "departure": "DUB",
      "departureDatetime": "2018-06-15T17:25:00",
      "destination": "WRO",
      "flightNumber": "FR153",
    }
  }
};

describe('FlightService', () => {
    it('Should return a list of flights', () => {
        let list = FlightService.list();
        expect(list).toMatchSnapshot();
    });
});

const user = {
  createdAt: new Date(),
  id: Math.floor(Math.random() * 20),
  name: 'LeBron James',
};

describe('User', () => {
    it('will fail every time', () => {
      	//L'id est généré de manière aléatoire, il changera à chaque nouveau test
		//idem pour la date
        //Créer un test snapshot qui failera à chaque fois


       // expect(user).toMatchSnapshot();
    });
});

const user_2 = {
  createdAt: new Date(),
  id: Math.floor(Math.random() * 20),
  name: 'LeBron James',
};

describe('User', () => {
    it('will check the matchers and pass', () => {
        //Adapter le test snapshot à l'aide des options disponibles
        //afin de parvenir à valider le test de l'user à chaque fois malgré 
        //le changement de l'id
        expect(user_2).toMatchSnapshot({
            createdAt: expect.any(Date),
            id: expect.any(Number),
          });
    });
});

const user_3 = {
  createdAt: new Date(),
  id: Math.floor(Math.random() * 20),
  name: Math.random().toString(36).substring(7),
};

describe('User', () => {
    it('will check the matchers and pass', () => {
        //Adapter le test snapshot à l'aide des options disponibles
        //afin de parvenir à valider le test de l'user à chaque fois malgré 
        //le changement de l'id
        const user = {
            createdAt: new Date(),
            name: 'Bond... James Bond',
          };
        
          expect(user).toMatchSnapshot({
            createdAt: expect.any(Date),
            name: 'Bond... James Bond',
          });
       
    });
});