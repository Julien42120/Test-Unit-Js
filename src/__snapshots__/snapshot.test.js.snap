// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightService Should return a list of flights 1`] = `
Object {
  "arrivalDatetime": "2018-06-15T21:00:00",
  "departure": "DUB",
  "departureDatetime": "2018-06-15T17:25:00",
  "destination": "WRO",
  "flightNumber": "FR153",
}
`;

exports[`User will check the matchers and pass 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`User will check the matchers and pass 2`] = `
Object {
  "createdAt": Any<Date>,
  "name": "Bond... James Bond",
}
`;
